async-nats/src/lib.rs

```diff
@@ -521,21 +454,104 @@ impl SubscriptionContext {
     }
 }
 
-/// A connector which facilitates communication from channels to a single shared connection.
-/// The connector takes ownership of the channel.
+///
+pub struct Connector {
+    server_addrs: Vec<ServerAddr>,
+    options: ConnectOptions,
+}
+
+impl Connector {
+    pub(crate) async fn connect(&mut self) -> Result<(ServerInfo, Connection), io::Error> {
+        loop {
```

Comment:
In case of error in every passed address (or the only one) it will endlessly loop over and over the vector without ever returning the error.

Reply:
Split it into `try_connect` and `connect`.
The former tries once cycle and the latter will loop until exhausted (currently no exhaustion).

Reply:
I think we need any stopper here that exhaust.
For now - a local variable that stops looping after 128 attempts or so?